mySqlDatabaseRemote/src/mySqlDriver2/SearchInventoryGUI.java

```diff
@@ -0,0 +1,146 @@
+    private JTextField searchPid, searchTitle, searchSid;  // Use Swing's JTextField instead of AWT's TextField
+    private JButton submitButton;    // Using Swing's JButton instead of AWT's Button
+    private int fieldselection = 0;
+    private String record;
+ 
+    // Constructor to setup the GUI components and event handlers
+    public SearchInventoryGUI() {
+      // Retrieve the content-pane of the top-level container JFrame
+      // All operations done on the content-pane
+      
+      Container cp = getContentPane();
+      cp.setLayout(new FlowLayout());
+      JPanel panel = new JPanel();
+      panel.setLayout(new FlowLayout());
+      panel.setVisible(true);
+
+      searchPid = new JTextField(10);
+      searchPid.setEditable(false);
+      searchTitle = new JTextField(10);
+      searchTitle.setEditable(false);
+      searchSid = new JTextField(10);
+      searchSid.setEditable(false);
+      panel.add(searchPid);
+      panel.add(searchTitle);
+      panel.add(searchSid);
+    
+      
+      pid = new JRadioButton("Product ID");
+      pid.addActionListener(new ActionListener() {
+        @Override
+        public void actionPerformed(ActionEvent e) {
+            setFieldselection(1);
+            searchPid.setEditable(true);
+            searchSid.setEditable(false);
+            searchTitle.setEditable(false);
+        }
+     });
+      cp.add(pid);
+
+      title = new JRadioButton("Product Title");
+      title.addActionListener(new ActionListener() {
+        @Override
+        public void actionPerformed(ActionEvent e) {
+            setFieldselection(2);
+            searchPid.setEditable(false);
+            searchSid.setEditable(false);
+            searchTitle.setEditable(true);
+        }
+     });
+      cp.add(title);
+
+      sid = new JRadioButton("Supplier's ID");
+      sid.addActionListener(new ActionListener() {
+        @Override
+        public void actionPerformed(ActionEvent e) {
+            setFieldselection(5);
+            searchPid.setEditable(false);
+            searchSid.setEditable(true);
+            searchTitle.setEditable(false);
+        }
+     });
+      cp.add(sid);
+      cp.add(panel);
+
+      ButtonGroup btnGp = new ButtonGroup();
+      btnGp.add(pid);
+      btnGp.add(title);
+      btnGp.add(sid);
+
+      
+
+ 
+
+
+
+      submitButton = new JButton("Submit");
+      submitButton.addActionListener(new ActionListener() {
+        @Override
+        public void actionPerformed(ActionEvent e) {
+           if(fieldselection == 1 && searchPid.getText() != null){
+               record = searchPid.getText();
+               
+           }
+           else if(fieldselection == 2 && searchTitle.getText() != null){
+               record = searchTitle.getText();
+           }
+           else if(fieldselection == 5 && searchSid.getText() != null){
+               record = searchSid.getText();
+            }
+            //System.out.println(record + " " + fieldselection);

```

Comment:
Do we need a dispose() here?

Reply:
I believe so. That way it closes but doesn't terminate the run